### PR TITLE
feat: nestjs-module 5.0.0 and logger 3.0.0

### PR DESCRIPTION
**Pull request by jmcdo29**
This PR brings many many updates to ogma including performance (thanks to @H4ad) and a new filter logger to log exceptions that the interceptor cannot catch. It also brings in support for NestJS v10 which was recently released! There are a couple of breaking changes worth mentioning that will already be mentioned in the CHANGELOG as well, but the big thing is that the options passed to the `OgmaModule` no longer need to include a `service` and `interceptor` section, and to register a parser for the `OgmaInterceptor` to use, all you need to do now is register a parser as a provider. There are a few format changes to the logs, namely in json mode letting the time be a number instead of a timestamp. There will probably be a few PRs to get this all set up just right to get everything deployed, especially due to changing package version

### CODE REVIEW

**Comment by jmcdo29** [comment]
@SocketSomeone @H4ad if y'all want to give this a review of any sort I'd appreciate it. I know there's a lot here to possibly review without full context, so even just the READMEs are appreciated. Just a second (or third) set of eyes before I release it :) Y'all are great!

File: .changeset/rich-llamas-relate.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+- time is now output in milliseconds since epoch in JSON format
+- JSON logs keep the `.message` property even in the case of logging objects
+- default stream in Node environments now uses `sonic-boom` instead of `process.stdout`
+
+- Unification of how objects get stringified to make all non-standard properties act the same.
+- Handle bigints. not much more to say about that
+- `sonic-boom` is a very fast stream implementation from pino that makes ogma close to comparable speed wise. We're also now using `fast-safe-stringify` to help speed up the stringification
+- speed. Using `toISOString()` is a bit slow. Fast enough for the string format but wanted to speed up JSON just a tiny bit more, plus it's supposed to be machine readable more than anything
+- By keeping the `.message` property, even in the case of objects we're able to eliminate an if check and keep the speeds up on the JSON logging. Also, gives a defined proeprty that the user defined message will **always** be at
+- As mentioned earlier, `sonic-boom` is fast. Blows `process.stdout` out of the water
```
Comment (H4ad) [inline .changeset/rich-llamas-relate.md]: Some of those messages look duplicated, like handle bigints and sonic boom, mention just one time should be enough.

File: .changeset/shy-beans-tap.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+console.log(style.blue.apply('Hello'));
+```
+
+### After
+
+```ts
+import { style } from '@ogma/styler';
+
+console.log(style.blue()apply('Hello'));
```
Comment (H4ad) [inline .changeset/shy-beans-tap.md]:
```suggestion
console.log(style.blue().apply('Hello'));
```

**Comment by H4ad** [comment]
@jmcdo29 I was reviewing the code and found something that maybe we could revert, I changed the styler to use `method` instead of the `getter` to make it faster, but actually since these getters are inside a class I can assume that the performance will be the same as calling a method, [according to this benchmark](https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/RESULTS-v19.md#property-getter-access), the performance is only slower when you use getter inside a object.

**Review by H4ad — CHANGES_REQUESTED** [review]

**Comment by jmcdo29** [comment]
> @jmcdo29 I was reviewing the code and found something that maybe we could revert, I changed the styler to use `method` instead of the `getter` to make it faster, but actually since these getters are inside a class I can assume that the performance will be the same as calling a method, [according to this benchmark](https://github.com/RafaelGSS/nodejs-bench-operations/blob/main/RESULTS-v19.md?rgh-link-date=2023-06-16T22%3A02%3A30Z#property-getter-access), the performance is only slower when you use getter inside a object.

Oh wow. Am I reading right that a class getter and a class method have the same performance here?

**Comment by H4ad** [comment]
@jmcdo29 Yup, you are right, when I create a local benchmark to check, I thought the object getter and class getter had the same performance, but much time later I discover that the class getter is as fast as methods.

**Comment by jmcdo29** [comment]
> @jmcdo29 Yup, you are right, when I create a local benchmark to check, I thought the object getter and class getter had the same performance, but much time later I discover that the class getter is as fast as methods.

Great to know. I guess we can revert all of styler 2.0.0. Or we can just keep it because that's a lot of work to undo and everything currently "works" with the new format. It's definitely nice to know they have the same performance though. Something to keep in mind for the future. 

Though, _techincally_ this change is harder to detect because `.apply()` is a standard JS method that can be called on any object. I had a few tests that initially did not change to the new format and got strange `Number.apply()` errors instead of styled strings.

Hmm, :thinking: yeah I think we'll actually revert that back and make it just logger and the nestjs-modules

File: .changeset/shy-beans-tap.md (hunk 2 above)
Comment (jmcdo29) [reply .changeset/shy-beans-tap.md]: Turns out I think we're gonna get rid of this changeset completely and keep the class getters, so this won't matter after that

**Pull request merged**